File: src/main/java/net/cazzar/mods/jukeboxreloaded/proxy/ClientProxy.java
package net.cazzar.mods.jukeboxreloaded.proxy;

import cpw.mods.fml.client.registry.ClientRegistry;
import cpw.mods.fml.common.network.FMLEmbeddedChannel;
import cpw.mods.fml.common.registry.VillagerRegistry;
import cpw.mods.fml.relauncher.Side;
import net.cazzar.mods.jukeboxreloaded.blocks.TileJukebox;
import net.cazzar.mods.jukeboxreloaded.blocks.render.tile.TileJukeboxRender;
import net.cazzar.mods.jukeboxreloaded.blocks.render.item.ItemJukeboxRender;
import net.minecraft.item.ItemBlock;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.client.MinecraftForgeClient;

public class ClientProxy extends CommonProxy {
    @Override
    public void initVillagers() {
        super.initVillagers();
        VillagerRegistry.instance().registerVillagerSkin(3000, new ResourceLocation("cazzar", "textures/mob/villager.png"));
    }

    @Override
    public void initRendering() {
        ClientRegistry.bindTileEntitySpecialRenderer(TileJukebox.class, new TileJukeboxRender());
        MinecraftForgeClient.registerItemRenderer(ItemBlock.func_150898_a(jukeBox), new ItemJukeboxRender());
    }

    @Override
    public FMLEmbeddedChannel getChannel() {
        return getChannelMap().get(Side.CLIENT);
    }
}

File: src/main/scala/net/cazzar/mods/jukeboxreloaded/proxy/ClientProxy.scala
package net.cazzar.mods.jukeboxreloaded.proxy

import net.cazzar.mods.jukeboxreloaded.Util._
import net.cazzar.mods.jukeboxreloaded.blocks.BlockJukebox
import net.minecraft.client.Minecraft
import net.minecraft.client.resources.model.ModelResourceLocation
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent

/**
 * Created by Cayde on 15/12/2014.
 */
class ClientProxy extends IProxy {
  override def getWorld = Option(Minecraft.getMinecraft.theWorld)

  override def postInit(e: FMLPostInitializationEvent): Unit = {
    super.postInit(e)

    Minecraft.getMinecraft.getRenderItem.getItemModelMesher.register(BlockJukebox, 0, new ModelResourceLocation("jukeboxreloaded:jukebox", "inventory"))
  }
}
